js/admin/aioseop-count-chars.js

```diff
@@ -3,69 +3,356 @@
  *
  * @since 2.9.2
  * @since 3.2.0 Moved to its own file.
+ * @since 3.3.0 Complete refactoring.
  */
 
+var pluginDirName;
+var currentPage;
 
-var aiosp_title_extra = parseInt( aioseop_count_chars.extraTitleLength, 10 );
+if ('undefined' !== typeof aioseopCharacterCounter) {
+	var aioseopCharacterCounterIsGutenberg = aioseopCharacterCounter.isGutenberg;
+	var aioseopCharacterCounterAutogenerateDescriptions = aioseopCharacterCounter.autogenerateDescriptions;
+	var extraTitleLength = +aioseopCharacterCounter.extraTitleLength;
+	currentPage = aioseopCharacterCounter.currentPage;
+	pluginDirName = aioseopCharacterCounter.pluginDirName;
+} else if ('undefined' !== typeof aioseopOGCharacterCounter) {
+	pluginDirName = aioseopOGCharacterCounter.pluginDirName;
+	currentPage = aioseopOGCharacterCounter.currentPage;
+}
 
-jQuery( document ).ready( function() {
-	aioseopInitCounting();
-});
+$(function () {
+	"use strict";
 
-/**
- * AIOSEOP Init Counting
- *
- * @since ?
- */
-function aioseopInitCounting(){
-	/* count them characters */
-	jQuery( '.aioseop_count_chars' ).on('keyup keydown', function(){
-		aioseopCountChars( jQuery(this).eq(0), jQuery(this).parent().find('[name="' + jQuery(this).attr('data-length-field') + '"]').eq(0));
-	});
-	jQuery( '.aioseop_count_chars' ).each(function(){
-		aioseopCountChars( jQuery(this).eq(0), jQuery(this).parent().find('[name="' + jQuery(this).attr('data-length-field') + '"]').eq(0));
-	});
-}
+	var inputField;
+	var cntField;
+	var fieldSize;
+	var aioseopCharacterCounterTimeout = 0;
 
-/**
- * @summary Counts characters.
- *
- * @since 1.0.0
- * @since 2.9.1 Fix JS conflict with LearnDash and function name.
- *
- * @param Object $field.
- * @param Object $cntfield.
- * @return Mixed.
- */
-function aioseopCountChars( field, cntfield ) {
-	var extra = 0;
-	var field_size;
-	if ( ( field.attr('name') === 'aiosp_title' ) && ( typeof aiosp_title_extra !== 'undefined' ) ) {
-		extra = aiosp_title_extra;
+	aioseopAddEventListeners();
+	// Fire once on page load.
+	aioseopCountChars();
```

Comment:
I've moved everything inside the $(function() {} ); statement. Had to add a small comment for eslint to ignore the amount of statements inside since it's just a wrapper.

Reply:
Oh, an anonymous function...yea, that would be done differently than what I commented.

The error you are getting is from the number of Variable Statements you have wrapped in that single function. I believe the limit is 10. The only recommendation I would have is to reduce the number of variables, but I'm ok with ignoring that report.

Reply:
It's not the amount of variables, but the amount of statements inside that triggers the warning.  The max. is 30 and there are about 35 at the moment, but since it's a wrapper we shouldn't be strict about following the coding standards. I had a quick look around online and looks like other people do the same thing, so good to go!